_posts/2024-03-25-whats-cooking.md

```diff
@@ -0,0 +1,224 @@
+
+Initially, we began by writing workflow files to automate the execution of tests with each push to the development branch on GitHub.
+
+Over time, this progressed into developing workflow files aimed at building and deploying our backend onto App Runner, thereby enabling seamless access to our backend without the necessity of local execution.
+
+Within our GitHub workflow files, we designed jobs dedicated to constructing Docker images using Maven, facilitating the subsequent pushing of the latest backend image to Amazon Elastic Container Registry (ECR).
+
+Following the image push to ECR, we employed Terraform to deploy this image onto App Runner.
+
+To streamline the orchestration of these jobs, which were authored in separate files for organizational clarity, we utilized a main workflow file to dictate the sequence in which the jobs were to be executed.
+
+In our pursuit of enhancing backend security on App Runner, we opted to integrate an API Gateway.
+
+This gateway serves as a protective layer, mandating that all frontend requests transit through it and ensuring access to the backend only if the appropriate key is present in the request headers.
+
+# Leveraging AI (DRAFT - Jonathan)
+
+In our application, the backend receives data via HTTP-request delivered by the Chrome plugin. It then carries out various operations on this data, with most of them leveraging AI.
+
+The first operation is cleaning the data by removing undesired elements.
+In our context, this means sifting out inedible ingredients from a collection of potentially edible ones.
+
+Once we have this refined list of ingredients, we move on to our second operation, which involves generating a recipe. The AI model must consider all user-provided recipe requirements and consistently format its responses. Failure to do so could result in the AI model's responses being challenging to reliably parse into objects in a Java environment.
+
+To address this issue, there are two potential solutions.
+
+The first approach entails creating a custom AI model, allocating sufficient resources, and training it to align with your specific use case. Although the setup process may demand more time and resources, it could streamline prompt messages and decrease the number of tokens used in AI operations.
+
+The second approach involves utilizing a standard AI model while supplying it with detailed prompt messages. This method simplifies the comparison of your AI model's out-of-the-box performance, a convenience we appreciated in our study. Given that prompts are stateless, we must furnish all necessary information, instructions, and context with each prompt. To ensure proper formatting of the AI model's responses, we implemented a prompting technique known as "few-shot prompting." Essentially, with few-shot prompting, you provide the AI model with an example question and answer from which it can discern a pattern. For this approach, we devised a template encompassing all required information.
```

Comment:
"This method simplifies the comparison of your AI model's out-of-the-box performance, a convenience we appreciated in our study." Wat bedoel je hier juist mee?

Reply:
De hele alinea zou het contrast moeten veruidelijken met "the first approach". Namelijk dat we met de Few-Shot Prompting techniek geen complexe training of geavanceerde configuratie nodig hebben om de AI output te sturen. Hiermee kunnen we dus eenvoudiger de verschillende AI modellen vergelijken in hun standaard configuratie, zonder dat ze specifiek getrained zijn op maaltijden, voeding of recepten.

Ik heb de alinea herschreven naar:
The second approach involves utilizing a standard AI model while supplying it with detailed prompt messages. **Using this method does not require complex training nor advanced configuration of AI models, allowing us to compare their out-of-the-box performance.** Given that prompts are stateless, we must furnish all necessary information, instructions, and context with each prompt.